controllers/metal3machine_controller.go

```diff
@@ -519,14 +505,6 @@ func (r *Metal3MachineReconciler) Metal3DataToMetal3Machines(obj client.Object)
 	return requests
 }
 
-// setError sets the ErrorMessage and ErrorReason fields on the metal3machine
```

Comment:
This is a generic function for setting error messages and should not be removed.
If error should not be set as is indicated by the title, then one can set to the fields to nil/"".

Reply:
Good point @Xenwar . Also if we are removing this function, there isn't any point to leave the 	`clearErrorM3Machine(capm3Machine)` function also?

Reply:
If it helps my case, there is a similar setup on AWS
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/a519d67fbcbd5e50b517576e06d578690f6b70a7/pkg/cloud/services/eks/cluster.go#L134-L142

Reply:
Thanks @Xenwar for pointing it out. I agree with you. I removed it because there is no use of it, and golinter doesn't allow to keep unused func(). I would personally would not set any error instead of setting nil, but I don't have strong opinion which way is better. As you suggested, we can set the message & error capi error type to nil. As such, no func() removal and no linter complains.

Reply:
@Xenwar @kashifest PTAL updated accordingly